src/logging.ts

```diff
@@ -174,12 +174,27 @@ export function createLoggingMiddleware({
   }
 }
 
+// Adapt the got options to a format that can be used by the default request serializer
+// See https://github.com/sindresorhus/got/blob/2b1482ca847867cbf24abde4d68e8063611e50d1/source/core/index.ts#L312
+function makeFakeRequestFromGotOptions(gotOptions: any) {
```

Comment:
should we add a test to ensure this doesn't crash for other libraries until we add support / decide to use a single library everywhere?

Reply:
+1 -- even though it looks like it handles any shape for `gotOptions` now, in the future someone could come along and tweak it in a way that might break it for other libraries.  E.g., add `gotOptional.headers.foo`.  seems like simple unit test that asserts `makeFakeRequestFromGotOptions` doesn't crash with an empty object or something would do the trick.

Reply:
Thanks, I added new tests with axios and node-fetch and uncovered a crash :D